model/compile/QtiAssetCompiler/XIncludeXmlInjector.php

```diff
@@ -38,18 +38,22 @@ class XIncludeXmlInjector extends ConfigurationService
      * @param DOMDocument $domDocument
      * @param PackedAsset[] $packedAssets
      */
-    public function injectSharedStimulus(DOMDocument $domDocument, array $packedAssets) : void
+    public function injectSharedStimulus(DOMDocument $domDocument, array $packedAssets): void
     {
+        $original = $domDocument->getElementsByTagName('include');
+        $elements = [];
+
         /** @var DOMElement $xincludeNode */
-        foreach ($domDocument->getElementsByTagName('include') as $xincludeNode) {
+        foreach ($original as $xincludeNode) {
+            $elements[] = $xincludeNode;
+        }
 
+        foreach ($elements as $xincludeNode) {
```

Comment:
Why this change? It is doing same as before, no?

Reply:
before: 
 `replaceChild` called on the iterable nodes collection made changes to it; That lead to fact internal pointer pointed to not correct element of collection and not all of them replaced in reality. Literally it was skipping even shared stimulues. 
after:
Copiying to simple array allowes to use simple iterations and avoide loosing elements

You can visualize it with an easy example - try to run delivery with 1 item, containing 5 shared stimuluses ( no matter withing 1 interaction or each of them can be wrapped into own )

Reply:
ok, I got. We were updating the same object inside the loop, thanks!